File: FSE_ProjectManager/PM.Utilities/Filter/FilterState.cs
using System.Collections.Generic;

namespace PM.Utilities.Filter
{
    public class FilterState
    {
        /// <summary>
        /// The number of records to be skipped by the pager.
        /// </summary>
        public int Skip { get; set; }

        /// <summary>
        /// The number of records to take.
        /// </summary>
        public int Take { get; set; }

        /// <summary>
        /// The descriptors used for sorting.
        /// </summary>
        public List<SortDescriptor> Sort { get; set; }

        /// <summary>
        /// The descriptors used for filtering.
        /// </summary>
        public CompositeFilterDescriptor Filter { get; set; }

        /// <summary>
        /// Time zone offset (minutes) with client browser.
        /// Only applicable to DateTime
        /// </summary>
        public double? TimeZoneOffset { get; set; }
    }
}

File: FSE_ProjectManager_UI/src/app/shared/filter/models/filter-state.ts
import { CompositeFilterDescriptor } from './composite-filter-descriptor';
import { SortDescriptor } from './sort-descriptor';

/**
 * The state of the data operations applied to the Grid component.
 */
export interface FilterState {
  /**
   * The number of records to be skipped by the pager.
   */
  skip?: number;
  /**
   * The number of records to take.
   */
  take?: number;
  /**
   * The descriptors used for sorting.
   */
  sort?: Array<SortDescriptor>;
  /**
   * The descriptors used for filtering.
   */
  filter?: CompositeFilterDescriptor;

  /** Time zone offset (minutes) with client browser.
   * Only applicable to DateTime
  */
  timeZoneOffset?: number;
}
